server/serverHelpers/serverUtilities.ts

```diff
@@ -106,6 +106,24 @@ const OPERATION_FUNCTION_MAP = initializeOperationFunctionMap();
 
 server.operation = operation;
 
+export type OperationDefinition = {
```

Comment:
@dawsontoth This an attempt at defining an operation in a way that should be conducive to a REST API. This allows the parameters to specify `in: 'path'` which will be useful for GET/PUT/DELETE methods where there is identifying parameter that logically could go in the path.

Reply:
@kriszyp what do you think about this simplified type? We can expand it more later for more generics, but it could be a start. We can add the schema stuff later when we know the shape that it will take, and I can improve the R and P generics at that point too to be easier to use.
```
export interface OperationDefinition<R = any, P = any> {
	name: string;
	execute: (params: P) => R | Promise<R>;
	httpMethod?: 'delete' | 'get' | 'head' | 'options' | 'patch' | 'post' | 'put' | 'trace';
	isJob?: boolean;
}
```